[솔로몬] Step3: plans 레이어

juchanhwang opened the pull request:
요구사항을 100퍼센트 충족하지는 못했지만, 나름? 구현이 되어서 pr보냅니다!
(비록 난잡한 코드지만) 수정해야할 부분과, 부족한 부분 피드백 부탁드립니다!

juchanhwang committed:
add: nav-plans-bar

juchanhwang committed:
add: plan-card html, css
card-main, card-bottom, logo-arrow-bottom 영역에 html, css 추가

juchanhwang committed:
1. add amazon.js file
2. add: nav-plan-bar css transition 효과 적용
3. add: js파일에 scroll event 사용

crongro commented:
커밋로그는 한글로 써보는것도 좋을거에요.
어떤 작업을 했는지  로그를 되돌려 봤을때 알 수 있도록 내용을 작성해보도록 하세요.

crongro commented on amazon.css:
```diff
@@ -230,14 +226,14 @@ header #header-bottom-nav {
 
 #header-bottom-nav #nav-right {
```
아이디는 페이지(html)에 하나뿐이죠.

따라서 이렇게만 하는게 맞습니다.

 #nav-right {
}

crongro commented on amazon.css:
```diff
@@ -371,11 +367,252 @@ header #header-bottom-nav {
+}
+
+.card {
+  /* border: solid 1px gray; */
```
안쓰는 코드는 삭제~~ 하는 습관

crongro commented on amazon.css:
```diff
@@ -371,11 +367,252 @@ header #header-bottom-nav {
+  height: 12px;
+  opacity: .64;
+  position: relative;
+  background-size: 400px 900px;
```
background 의 다양한 속성을 잘 알아두면 좋아요~~
이미지 배치할때 잘 사용됩니다. 
👍

crongro commented on amazon.html:
```diff
@@ -113,14 +113,115 @@
+              <div class="card-content"> FREE Same-Day Delivery </div>
+            </div>
+          </div>
+          <div class="card-list" id="card-list2">
```
card list 라고 했는데요.
그렇다면 list tag를 사용하는 게 좀더 의미가 맞을 거 같네요.

crongro commented on amazon.js:
```diff
@@ -0,0 +1,23 @@
+let logoArrow = document.querySelector(".logo-arrow");
```
amazon.js 코드안은 모두 전역변수인데요.

이걸 함수를 좀 활용하거나, class를 활용하면 좋겠어요.

```javascript
 class ScrollRunner {
  constructor() {
    this.init();
  }

  init() {
     let a = document.querySelector("ddd");
    a.addEventListener("click", ()=>this.handler() );
  }
   handler(e) {.....}
}

```

이런식으로요~

crongro reviewed (changes requested):
수고햇어요~
어려운 css를 많이 이해하게 됐네요.

amazon.js는 모듈화(함수나 객체화)를 하면 좋겠어요.
수정하고 다시 보내주세요~

juchanhwang committed:
amazon.js파일 수정
1. window영역에서 작성한 코드를 클래스로 분리
1-1. scroll이 발생하는 event는 PlanBarScrollEvent로, click이 발생하는 event는 PlanCardClickEvent로 나눔

juchanhwang committed:
plan-card의 하위 태그들의 class명 수정
card-content-main태그를 div->ul 태그로 변경

crongro reviewed (approved):
ES6 class를 만들때, 즉 객체를 만들때는 그 객체가 하나의 역할을 하는지에 대해서 늘 고민하세요~
판단하기 어려우면 너무 잘게 나누기보다, 하나의 클래스에 어울리는 일들을 모아두면됩니다.
그러다가 아니다 싶으면~ 객체(클래스)를 분리하고요.

지금 아래 두 개의 Class는 합쳐도 괜찮을거 같아요.

css수정도 잘하셨어요~
이제 JS 개발에 집중해봅시다!
